app/main.py

```diff
@@ -2,5 +1,19 @@
+from typing import Callable, Any
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_storage = {}
+
+    @wraps(func)
+    def wrapper(*args) -> Any:
+        if args in cache_storage:
+            print("Getting from cache")
+            return cache_storage[args]
+        else:
+            print("Calculating new result")
+            result = func(*args)
+            cache_storage[args] = result
```

Comment:
Similar to the previous comment, ensure that the arguments used as keys in the cache_storage dictionary are hashable to avoid runtime errors.